### DISCUSSION TITLE
Option to "discard" changes for new unsaved entry

### DISCUSSION THREAD

**Discussion by richhayler** [craftcms/cms#11482] [Ideas] [answered]
For the scenario where you create a new entry, click into the title field and then navigate away from the entry edit page (resulting in an unintended "untitled entry" from being created) - would it make sense to display the "discard" button which when clicked prevents the "untitled entry" from being created?

This would prevent entries from being created unintentionally and save an extra couple of steps (if you chose to remove the entry) of having to select the entry in the entries listing page and then click on the dropdown to select the "delete" option.

<img width="233" alt="CleanShot 2022-06-22 at 22 53 37@2x" src="https://user-images.githubusercontent.com/2505643/175035271-25186c76-7071-4d72-8113-f039738a6d7b.png">

<img width="421" alt="CleanShot 2022-06-22 at 22 48 47@2x" src="https://user-images.githubusercontent.com/2505643/175035245-369afaac-614b-48b9-971b-3e9c78e9b59c.png">

While I am still not a fan of the way the "discard" button reveals itself and shifts the edit form interface down, it makes sense to me from a consistency and functional perspective to display it in this use case.

I appreciate this means that the "discard" button would then get displayed every time a new entry was created (and the edit form interacted with), but maybe this would also encourage further consideration for how this functionality could be improved so it achieves its goal of drawing attention to the fact that changes are not saved, aids the author with the option to discard their changes but does not  disrupt the authoring flow in the process.

On the flip side for venteran authors would are fully aware their changes are unsaved (don't need to be reminded) and have no intention to discard them, the shifting edit form, while a minor quibble is respectfully a point of frustration.

**Comment by brandonkelly** [comment, marked as answer]
This would be inconsistent with how the Discard buttons currently work.

They currently only discard any unsaved _changes_ to the entry, whereas in this case it would end up deleting the entire entry. I suspect many people would end up losing content if they pressed it for an unpublished draft, thinking only their most recent changes would be discarded.

Plus, there’s already a quick way to delete an unpublished draft if you decide you didn’t want to keep it: the “Delete draft” action, which lives in the exact same place that “Delete entry” lives for fully saved entries.

<img src="https://user-images.githubusercontent.com/47792/175046545-5c347c1e-7173-4312-82f2-4b91dd86afdb.png" width="282" height="236" alt="The action menu on the edit page for an unpublished draft, with the “Delete draft” option highlighted">

> For the scenario where you create a new entry, click into the title field and then navigate away from the entry edit page (resulting in an unintended "untitled entry" from being created)

To clarify, the entry should only get autosaved once you’ve actually made a change to it. If it’s getting autosaved simply by clicking on the Title field (and nothing more), that would be a bug worth looking into separately.
